### DISCUSSION TITLE
New JKBMS "JK_DZ11B1A24S" (balancer only)

### DISCUSSION THREAD

**Discussion by gerold1234567** [Louisvdw/dbus-serialbattery#578] [Q&A] [answered]
Good day,

I organized a new JKBMS. I cannot connect to VenusOS (cerbo).
Specifications : Hw.V: V11.XW / Sw.V: V11.26
Is it possible that this BMS is not yet able to communicate with the dbus-serialbattery.

Best regards
Gerold

![20230425_173948](https://user-images.githubusercontent.com/131859609/234521753-15544b6c-0b30-4456-a121-9cfd5f8ff362.jpg)

![2023-04-26 um 08 09 21](https://user-images.githubusercontent.com/131859609/234522208-6aee082a-588c-41ff-b803-c66115397f1f.jpg)

**Comment by mr-manuel** [comment]
Hello,

it should be possible. I saw already some to connect a V11.X JKBMS. Have you done some troubleshooting? Please look at the wiki for that.

**Comment by gerold1234567** [comment]
Good day,

I have dealt with the WIKI in advance. I would like to ask you if I understood and implemented it correctly.

1. Command: nano /data/log/serial-starter/current at 04/28/23 07:27

@400000006448ec2607dd4c4c INFO: Start service gps-dbus.ttyUSB0 once
@400000006448ec2f26a09a6c INFO: Start service vedirect-interface.ttyUSB0 once
@400000006448ec3504d1f87c INFO: Start service dbus-cgwacs.ttyUSB0 once
@400000006448ec3a1d4412b4 INFO: Start service dbus-fzsonick-48tl.ttyUSB0 once
@400000006448ec3d30feee3c INFO: Start service dbus-imt-si-rs485tc.ttyUSB0 once
**@400000006448ec410d384124 INFO: Start service dbus-modbus-client.serial.ttyUSB0 once** <- runs on USB 0 (that's how I would understand it?)
@400000006448ec482ae5124c INFO: Start service dbus-serialbattery.ttyUSB0 once
@400000006448ec5417e23c24 INFO: Start service gps-dbus.ttyUSB0 once"

2. Command: tail -f /data/log/serial-starter/current | tai64nlocal at 04/28/23 07:30

root@einstein:/data/log# tail -f /data/log/serial-starter/current | tai64nlocal
2023-04-28 05:34:10.580289500 INFO: Start service dbus-modbus-client.serial.ttyUSB0 once
2023-04-28 05:34:18.053774500 INFO: Start service dbus-serialbattery.ttyUSB0 once
2023-04-28 05:34:29.694804500 INFO: Start service gps-dbus.ttyUSB0 once
2023-04-28 05:34:39.162206500 INFO: Start service vedirect-interface.ttyUSB0 once
2023-04-28 05:34:44.629695500 INFO: Start service dbus-cgwacs.ttyUSB0 once
2023-04-28 05:34:50.010033500 INFO: Start service dbus-fzsonick-48tl.ttyUSB0 once
2023-04-28 05:34:53.315843500 INFO: Start service dbus-imt-si-rs485tc.ttyUSB0 once
2023-04-28 05:34:56.858713500 INFO: Start service dbus-modbus-client.serial.ttyUSB0 once
**2023-04-28 05:35:04.282464500 INFO: Start service dbus-serialbattery.ttyUSB0 once** <- (when plugging in and unplugging it stops and starts the process, is that correct?)
2023-04-28 05:35:15.899887500 INFO: Start service gps-dbus.ttyUSB0 once
2023-04-28 05:35:25.355795500 INFO: Start service vedirect-interface.ttyUSB0 once
2023-04-28 05:35:30.842149500 INFO: Start service dbus-cgwacs.ttyUSB0 once

3. Command: tail -f /data/log/dbus-serialbattery.ttyUSB0/current | tai64nlocal at 04/28/23 07:34

root@einstein:/data/log# tail -f /data/log/dbus-serialbattery.ttyUSB0/current | tai64nlocal
2023-04-28 05:35:13.202699500 ERROR:SerialBattery:>>> ERROR: No reply - returning
2023-04-28 05:35:13.203684500 INFO:SerialBattery:Testing Lifepower
2023-04-28 05:35:13.526988500 ERROR:SerialBattery:>>> ERROR: No reply - returning
2023-04-28 05:35:13.528150500 ERROR:SerialBattery:>>> ERROR: Incorrect Data
2023-04-28 05:35:13.530160500 INFO:SerialBattery:Testing Renogy
2023-04-28 05:35:13.834737500 ERROR:SerialBattery:>>> ERROR: No reply - returning
2023-04-28 05:35:13.835840500 INFO:SerialBattery:Testing Renogy
2023-04-28 05:35:14.136620500 ERROR:SerialBattery:>>> ERROR: No reply - returning
2023-04-28 05:35:14.140883500 INFO:SerialBattery:Testing Ecs
2023-04-28 05:35:14.703485500 ERROR:SerialBattery:ERROR >>> No battery connection at /dev/ttyUSB0

and here I see that there is no connection to a BMS (is that correct?)

Can you conclude that the usb adapter is not working (I swapped channels A and B, no success)

I measure the BMS RS485/can socket from GND to (pin A/H) +5 V and (pin B/L) +0 V.
So I assume that it is an RS485 interface?

>Herr Manuel
> Hello,
>
> it should be possible. I've already seen quite a bit to connect a V11.X JKBMS. Did you do some troubleshooting? Please see the wiki for more information.

Yesterday I searched various forums to see if anyone had a new BMS running, I couldn't find anything with this connection?


Best regards
Gerold

![Connection RS485CAN JK BMS](https://user-images.githubusercontent.com/131859609/235072328-bbbca13c-8627-4ef1-9a4f-a1f28e6a874b.jpg)

![Connection 2 x RS485CAN JK BMS](https://user-images.githubusercontent.com/131859609/235072359-c89dc0ec-f909-4f3d-bd7c-89dedc701f8b.jpg)

**Reply by mr-manuel** [reply]
Normally the RS485/CAN port of a JKBMS is not RS485 or TTL, but only CAN. See https://github.com/Louisvdw/dbus-serialbattery/discussions/250#discussioncomment-4452179

But you have a different BMS and from the pictures I saw, that you tried with a RS485 and TTL adapter, am I right?

Please ask your dealer what signal exactly come out of this port.

**Reply by gerold1234567** [reply]
Hello

>But you have another BMS and from the pictures I saw you tried RS485 and TTL adapter am I right? **yes i used**

>Please ask your dealer which signal is coming out of this connector.
Does the document help you?

[JK-DZ11B2A24S equalizer RS485 communication protocol V1 3.pdf](https://github.com/Louisvdw/dbus-serialbattery/files/11350861/JK-DZ11B2A24S.equalizer.RS485.communication.protocol.V1.3.pdf)

**Reply by mr-manuel** [reply]
Unfortunately not, since it's all chinese :-D

**Reply by gerold1234567** [reply]
> Unfortunately not, since it's all chinese :-D

[JK RS 485 Protokoll EN.pdf (Page1-6).pdf](https://github.com/Louisvdw/dbus-serialbattery/files/11353247/JK.RS.485.Protokoll.EN.pdf.Page1-6.pdf)

I translated with my cell phone :-)

**Reply by mr-manuel** [reply]
Are there no other ports on the BMS? I fear, that this BMS has no real RS485/TTL port.

Where did you bought the BMS? The only option here is to ask the seller, if this BMS have a real RS485 or TTL port.

**Reply by gerold1234567** [reply]
> Are there no other ports on the BMS? I fear, that this BMS has no real RS485/TTL port.
> 
> Where did you bought the BMS? The only option here is to ask the seller, if this BMS have a real RS485 or TTL port.

Bought here: 
https://german.alibaba.com/product-detail/B1A24S-RS485-JK-Smart-BALANCER-12S-1600756085897.html


According to the seller it should work:
![image](https://user-images.githubusercontent.com/131859609/235598256-3ea45892-b17d-4e49-86d2-f9dcb4dcfd1f.png)

**Reply by mr-manuel** [reply]
Ok, so the cables should be connected like:
GND -> GND
A -> A
B -> B

Only the RS485 will work. Are the TXD and RXD lights flashing?

The USB to TTL adapter (transparent one) will not work.

Could you post all settings in your JKBMS app? I remember that with newer BMS there is like an RS485 ID you can set.

Could you also post the complete output from `INFO:SerialBattery:Starting dbus-serialbattery` to the next `INFO:SerialBattery:Starting dbus-serialbattery`?

```bash
tail -f /data/log/dbus-serialbattery.*/current | tai64nlocal
````

**Reply by gerold1234567** [reply]
> Ok, so the cables should be connected like: GND -> GND A -> A B -> B
**> G-GND, A-TXD, B-RXD**

> Only the RS485 will work. Are the TXD and RXD lights flashing?
**> TXD = flashing, RXD = Lights up and does not flash**

> The USB to TTL adapter (transparent one) will not work.
**> I tested three RS 485 adapters (see pictures)**

> Could you post all settings in your JKBMS app? I remember that with newer BMS there is like an RS485 ID you can set.
**> Device Addr.: I have set 21 (see pictures)**

> Could you also post the complete output from `INFO:SerialBattery:Starting dbus-serialbattery` to the next `INFO:SerialBattery:Starting dbus-serialbattery`?
> 
> ```shell
> tail -f /data/log/dbus-serialbattery.*/current | tai64nlocal
> ```
[Neues Textdokument.txt](https://github.com/Louisvdw/dbus-serialbattery/files/11371376/Neues.Textdokument.txt)



![20230502_095136](https://user-images.githubusercontent.com/131859609/235614894-08132fb8-28f6-4aa1-8044-5ed53ee91380.jpg)
![20230502_095450](https://user-images.githubusercontent.com/131859609/235614922-caac419e-e670-45bb-89c7-5021810eed82.jpg)

![Screenshot_20230502_09
![Screenshot_20230502_092325](https://user-images.githubusercontent.com/131859609/235614830-0d1863a7-eed1-43d3-8d8f-c4a729083548.jpg)
2301](https://user-images.githubusercontent.com/131859609/235614816-ad7a1060-3e72-4c31-9b35-dfda0052eacd.jpg)
![Screenshot_20230502_092420](https://user-images.githubusercontent.com/131859609/235614853-df389f3c-6cf2-449f-a8ad-1c4097c51df8.jpg)
![Screenshot_20230502_092451](https://user-images.githubusercontent.com/131859609/235614873-0d06f49b-f467-4f0f-93ce-b91fc5119049.jpg)

**Reply by mr-manuel** [reply]
Please try to set the `device addr` to 1, 0 or 2 in the JKBMS settings. I think it should work with 1.

See also https://github.com/Louisvdw/dbus-serialbattery/issues/142.

**Reply by gerold1234567** [reply]
> `device addr`Bitte versuchen Sie in den JKBMS-Einstellungen den auf 1, 0 oder 2 zu setzen . Ich denke mit 1 sollte es funktionieren.
> 
> Siehe auch [Nr. 142](https://github.com/Louisvdw/dbus-serialbattery/issues/142) .

does not work too

**Reply by mr-manuel** [reply]
Unfortunately I have no other idea then. You could connect via Bluetooth. See https://github.com/Louisvdw/dbus-serialbattery/discussions/485

**Reply by gerold1234567** [reply]
> Unfortunately I have no other idea then. You could connect via Bluetooth. See #485

Okay thanks!
How many JK balancers can you connect with Bluetooth?

Would like to monitor 10 to 30 battery packs.
![20230210_173746](https://user-images.githubusercontent.com/131859609/235722102-273cb1e6-8cec-4a63-ac91-3bc790573f8c.jpg)
![20230307_161323](https://user-images.githubusercontent.com/131859609/235722155-8e97fd33-fa60-4457-bb2a-e72a6bd934d1.jpg)

**Reply by mr-manuel** [reply]
What are you doing with that much battery packs? Are you building some for others or testing? Are you using them in extreme conditions since you are using LTO batteries? :-)

[Here](https://louisvdw.github.io/dbus-serialbattery/faq/#how-many-usb-to-serial-adapters-can-i-connect) you can find how much serial USB adapter you can connect.

You would be the perfect person to test, how many Bluetooth BMS you can connect. You have to monitor CPU load and response time of the battery, to see how much the system can handle.

A Raspberry Pi 4 would maybe be better for handling that much BMS.

**Reply by gerold1234567** [reply]
> What are you doing with that much battery packs? Are you building some for others or testing? Are you using them in extreme conditions since you are using LTO batteries? :-)

I received this battery backete from a company that is in bankruptcy.
They don't have a balancer built in!
That's why the battery backet broke after about a year.
And I'm installing balancers now.
LTO are amazing, they can take a lot!
Now I'm building this battery backete just for me, maybe it should work for friends too.

> [Here](https://louisvdw.github.io/dbus-serialbattery/faq/#how-many-usb-to-serial-adapters-can-i-connect) you can find how much serial USB adapter you can connect.

That's why I only bought two at first. If I know which ones work or which method I use (RS-485, Bluetooth or CAN)? I'll buy the rest
I'm still testing which balancer I should take so that I can automatically monitor each cell?

> You would be the perfect person to test, how many Bluetooth BMS you can connect. You have to monitor CPU load and response time of the battery, to see how much the system can handle.
> 
> A Raspberry Pi 4 would maybe be better for handling that much BMS.

Everything currently runs on Cerbo GX

![20230425_174005](https://user-images.githubusercontent.com/131859609/235752016-102c8332-a588-4c9a-810d-152bda772073.jpg)

**Reply by mr-manuel** [reply]
From what I know, you get the most data out of a JKBMS. In this case, since you have so many battery banks, you maybe can make more pressure on the supplier.

Write me on Discord, maybe I find something, if I check directly on your device.

**Comment by gerold1234567** [comment]
Hello Manuel,

I managed to connect via Bluetooth.
Do not want to use the BMS control (DVCC).
Have no information watts and amps of the battery (see photo) Why is that?

PS: I changed the config file (see attachment)


![b2](https://user-images.githubusercontent.com/131859609/235882790-c8284876-a70c-408c-a43b-f0f68932fc79.JPG)
![b1](https://user-images.githubusercontent.com/131859609/235882797-ab3f1606-0e79-4bad-bf99-42839a56ea4b.JPG)

[data_log_dbus-blebattery_current_tai64nlocal.txt](https://github.com/Louisvdw/dbus-serialbattery/files/11381280/data_log_dbus-blebattery_current_tai64nlocal.txt)
[utils.txt](https://github.com/Louisvdw/dbus-serialbattery/files/11381282/utils.txt)
[config.default.txt](https://github.com/Louisvdw/dbus-serialbattery/files/11381283/config.default.txt)

**Reply by mr-manuel** [reply]
Did you change something in the `utils.py`? I think best would be, if you contact me on Discord or you give me a possibility to contact you directly. So we can troubleshoot, it the serial connection is sending data, but the driver isn't reading it yet and why the Bluetooth driver is not showing all data. This would save a lot of time ;-)

**Reply by mr-manuel** [reply]
> Do not want to use the BMS control (DVCC).

See the FAQ for this: [Can the driver be used as monitor only?](https://louisvdw.github.io/dbus-serialbattery/faq/#can-the-driver-be-used-as-monitor-only)

**Comment by mr-manuel** [comment, marked as answer]
I realized that this is no BMS, but only a balancer. So it has another protocol. Over Bluetooth it works, since it's using the same app.
